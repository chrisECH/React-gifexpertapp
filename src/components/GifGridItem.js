import React from 'react';
import PropTypes from 'prop-types'; //impt
import './styles/GifGridItem.css';

export const GifGridItem = ({title, url}) => {
   
    return (
        <div className="card animate__animated animate__fadeIn  animate__delay-1s">
            <img src={url} alt={title}></img>
            <p >{title}</p>

            
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}

