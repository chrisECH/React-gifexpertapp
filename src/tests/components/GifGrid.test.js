import { shallow } from "enzyme";
import '@testing-library/jest-dom';
import { GifGrid } from "../../components/GifGrid"
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');

  describe('Prueba para el <GrifGrid />', () => {

    const category = 'Vikingos';

    test('debe de hacer match con el snapshot', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper =shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se cargan las imagenes useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'cualquier cosa'
        },{
            id: '123',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'cualquier cosa'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        
        const wrapper =shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

    });
      
  });