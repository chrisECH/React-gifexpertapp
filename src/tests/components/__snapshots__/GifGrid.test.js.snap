// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba para el <GrifGrid /> debe de hacer match con el snapshot 1`] = `
<Fragment>
  <h3>
    Vikingos
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Prueba para el <GrifGrid /> debe de mostrar items cuando se cargan las imagenes useFetchGifs 1`] = `
<Fragment>
  <h3>
    Vikingos
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="cualquier cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="cualquier cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;
